src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedNewTests.cs

```diff
@@ -925,6 +925,94 @@ public static void Main()
                 );
         }
 
+        [Fact]
+        public void TestOverloadResolution2()
+        {
+            var comp = CreateCompilation(@"
+using System;
+class A
+{
+    public A(int i) {}
+}
+class B : A
+{
+    public B(int i) : base(i) {}
+}
+
+class Program
+{
+    static void M(A a) => Console.Write(""A"");
+    static void M(B a) => Console.Write(""B"");
+
+    public static void Main()
+    {
+        M(new(43));
+    }
+}
+", options: TestOptions.ReleaseExe);
+
+            CompileAndVerify(comp, expectedOutput: "B");
```

Comment:
@gafter I suspect this shouldn't work. (out var doesn't)

Reply:
also tagging @jcouv, in the late-filter, this should result in ambiguous call, right?

Reply:
Can either of you guys confirm this behavior? If we do betterness I think it defeats the purpose of late-filter..

Reply:
`default` works in this scenario, so I'd think target-typed `new` should too. We'd have to drill into the reason why out var doesn't.

Reply:
>We'd have to drill into the reason why out var doesn't.

I think the condition in `BetterConversionFromExpression` is responsible for that,

> ```cs
> // Neither conversion from expression is better when the argument is an implicitly-typed out variable declaration.
> ```

Interestingly, while we do betterness for default, that doesn't mean it contributes to overload resolution,
```cs
class Program
{
    static void M(Base a, int i) => Console.Write("A");
    static void M(Derived a, object i) => Console.Write("B");
    
    public static void Main()
    {
        M(default, 1); // ERROR: ambig call
    }
}
```
That'll be a good test case if we want to retain `default` behavior for `new()`.